### DISCUSSION TITLE
HINTS and STRUGGLES

### DISCUSSION THREAD

**Discussion by derekscholes** [USU-ITLS-x265-23-summer/class-repo#43] [Week 2 - Discussions]
Honestly, this week was very difficult for me. I really struggled with some of the Exercism activities, especially the last one. I felt like there were almost no instructions on how to solve it and was more trial and error. I felt like no excercises were given to help complete these assignments. As I look at week 3, I see that some of those assignments would have helped with the coding in the exercism assignments. When I really struggled, what helped most was watching explanations of the assignments by others and tried solving before I continued. Once I finished I then rewatched to make sure that I understood the concepts before moving on. 

Here is the playlist I used. https://www.youtube.com/watch?v=Xgrm-5_2mEA&list=PLJO7vAdVP8x1XreVWnTEtySN9BE8-Tr4o 

The FreeCodeCamp assignment was fairly straight forward. I dont think I had too many questions. It was also nice learning a bit about HTML and comparing to code on other websites.

In order to change the approach, I would like to have a better explanation of what the exercise entails. Half the time I didn't know what was required of me and it took me hours to find out what was being asked. It would have also been nice to do the JavaScript Algorithms and Data Structures in FreeCodeCamp prior to doing any of the exercism activities.

**Comment by jakegros** [comment]
@derekscholes great find for the playlist.  I got stuck easily with the Exercism projects.  As you mentioned, the lack of explanation--at least not knowing what is required or what it means--made it very difficult to even know how to go about the errors.  I found this [JS cheat sheet](https://www.interviewbit.com/javascript-cheat-sheet/#js-fundamentals) that I like because it describes what a lot of things mean and how to use.

**Comment by kassyoveson** [comment]
I am writing this after seeing your final project. You have come so far this semester and are a coding rock star now! This earlier post just proves how far you have come. Great job!
